internal/pkg/mail/repository/postgesql.go

```diff
@@ -0,0 +1,126 @@
+package repository
+
+import (
+	"context"
+	"liokor_mail/internal/pkg/common"
+	"liokor_mail/internal/pkg/mail"
+	"time"
+)
+
+type PostgresMailRepository struct {
+	DBInstance common.PostgresDataBase
+}
+
+func (mr *PostgresMailRepository) GetDialoguesForUser(username string, limit int, last int) ([]mail.Dialogue, error) {
+	rows, err := mr.DBInstance.DBConn.Query(
+		context.Background(),
+		"SELECT d.id, "+
+			"CASE WHEN d.user_1=$1 THEN d.user_2 WHEN d.user_2=$1 THEN d.user_1 END AS email, "+
+			"u.avatar_url, m.body, m.received_date FROM dialogues d JOIN mails m ON d.last_mail_id=m.id "+
+			"LEFT JOIN users u ON "+
+			"CASE WHEN d.user_1=$1 THEN SPLIT_PART(d.user_2,'@liokor.ru', 1)=u.username WHEN d.user_2=$1 THEN SPLIT_PART(d.user_1,'@liokor.ru', 1)=u.username END "+
+			"WHERE (d.user_1=$1 OR d.user_2=$1) AND d.id > $3 "+
+			"ORDER BY d.received_date DESC LIMIT $2;",
+		username,
+		limit,
+		last,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	dialogues := make([]mail.Dialogue, 0, 0)
+	for rows.Next() {
+		dialogue := mail.Dialogue{}
+		err = rows.Scan(
+			&dialogue.Id,
+			&dialogue.Email,
+			&dialogue.AvatarURLDB,
+			&dialogue.Body,
+			&dialogue.Received_date,
+		)
+		if err != nil {
+			return nil, err
+		}
+		//это костыль, который я потом исправлю
```

Comment:
Костыль заключается в пустой аватарке?

Если делаешь пометку "на потом", то хотя бы так:
```suggestion
		// TODO: это костыль, который я потом исправлю
```